pkg/converter/pod_converter.go

```diff
@@ -0,0 +1,54 @@
+package converter
+
+import (
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	k8sApiV1 "k8s.io/client-go/pkg/api/v1"
+	"reflect"
+)
+
+// Label which represents the namespace a given pod belongs to.
+const k8sNamespaceLabel = "calico/k8s_ns"
+
+type podConverter struct {
+}
+
+// NewPodConverter Constructor for podConverter
+func NewPodConverter() Converter {
+	return &podConverter{}
+}
+func (p *podConverter) Convert(k8sObj interface{}) (interface{}, error) {
+
+	if reflect.TypeOf(k8sObj) != reflect.TypeOf(&k8sApiV1.Pod{}) {
+		log.Fatalf("can not convert object %#v to workloadEndpoint. Object is not of type *v1.Pod", k8sObj)
+	}
+
+	pod := k8sObj.(*k8sApiV1.Pod)
+	endpoint := api.NewWorkloadEndpoint()
+	labels := make(map[string]string)
```

Comment:
Why do we need this intermediate? Can't we just populate `endpoint.Metadata.Labels` directly and save ourselves a variable?